compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -869,6 +894,36 @@ object desugar {
     }
   }
 
+  /** Given tpe parameters `Ts` (possibly empty) and a leading value parameter `(x: T)`,
+   *  map a method definition
+   *
+   *     def foo [Us] paramss ...
+   *
+   *  to
+   *
+   *     <extension> def foo[Ts ++ Us](x: T) parammss ...
+   *
+   *  If the given member `mdef` is not of this form, flag it as an error.
+   */
+
+  def makeExtensionDef(mdef: Tree, tparams: List[TypeDef], leadingParams: List[ValDef]) given (ctx: Context): Tree = {
+    val allowed = "allowed here, since collective parameters are given"
+    mdef match {
+      case mdef: DefDef =>
+        if (mdef.mods.is(Extension)) {
+          ctx.error(em"No extension method $allowed", mdef.sourcePos)
+          mdef
+        }
+        else cpy.DefDef(mdef)(tparams = tparams ++ mdef.tparams, vparamss = leadingParams :: Nil)
```

Comment:
huh, doesn't this miss the vparamss in mdef itself ?

Reply:
You are right.